analyses/oncoprint-landscape/01-plot-oncoprint.R

```diff
@@ -0,0 +1,241 @@
+
+# Get `magrittr` pipe
+`%>%` <- dplyr::`%>%`
+
+#### Directories and Files -----------------------------------------------------
+
+# Detect the ".git" folder -- this will in the project root directory.
+# Use this as the root directory to ensure proper execution, no matter where
+# it is called from.
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+
+# Path to the data obtained via `bash download-data.sh`.
+data_dir <- file.path(root_dir, "data")
+
+# Path to output directory for plots produced
+plots_dir <-
+  file.path(root_dir, "analyses", "oncoprint-landscape", "plots")
+
+if (!dir.exists(plots_dir)) {
+  dir.create(plots_dir)
+}
+
+# Source the color palette for plots
+source(
+  file.path(
+    root_dir,
+    "analyses",
+    "oncoprint-landscape",
+    "util",
+    "oncoplot-palette.R"
+  )
+)
+
+#### Command line options ------------------------------------------------------
+
+# Declare command line options
+option_list <- list(
+  optparse::make_option(
+    c("-m", "--maf_file"),
+    type = "character",
+    default = NULL,
+    help = "file path to MAF file that contains snv information",
+  ),
+  optparse::make_option(
+    c("-c", "--cnv_file"),
+    type = "character",
+    default = NULL,
+    help = "file path to SEG file that contains cnv information"
+  ),
+  optparse::make_option(
+    c("-f", "--fusion_file"),
+    type = "character",
+    default = NULL,
+    help = "file path to file that contains fusion information"
+  ),
+  optparse::make_option(
+    c("-g", "--goi_list"),
+    type = "character",
+    default = NULL,
+    help = "file path to file that contains list of genes to include on
+            oncoprint"
+  ),
+  optparse::make_option(
+    c("-p", "--png_name"),
+    type = "character",
+    default = NULL,
+    help = "oncoprint output png file name"
+  )
+)
+
+# Read the arguments passed
+opt_parser <- optparse::OptionParser(option_list = option_list)
+opt <- optparse::parse_args(opt_parser)
+
+maf <- opt$maf_file
+
+# Define cnv_file object here as it still needs to be defined for the `read.maf`
+# function, even if it is NULL
+cnv_file <- opt$cnv_file
+
+#### Read in data --------------------------------------------------------------
+
+# Read in metadata
+metadata <-
+  readr::read_tsv(file.path(data_dir, "pbta-histologies.tsv"))
+
+# Rename for maftools function
+metadata <- metadata %>%
+  dplyr::rename("Tumor_Sample_Barcode" = "Kids_First_Biospecimen_ID")
+
+# Read in MAF file
+maf_df <- data.table::fread(maf, stringsAsFactors = FALSE)
+
+# Read in cnv file
+if (!is.null(opt$cnv_file)) {
+  cnv_file <- data.table::fread(opt$cnv_file, stringsAsFactors = FALSE)
+  # TODO: Filter and set up `cnv_file` to be in the column format -
+  # "Hugo_Symbol, Tumor_Sample_Barcode, Variant_Classification" as required by
+  # the `read.maf function`
+}
+
+# Read in fusion file
+if (!is.null(opt$fusion_file)) {
+  fusion_file <- data.table::fread(opt$fusion_file, stringsAsFactors = FALSE)
```

Comment:
```suggestion
  fusion_file <- data.table::fread(opt$fusion_file, data.table = FALSE, stringsAsFactors = FALSE)
```
probably will make things a bit easier